pkg/apiserver/vshn/nextcloud/vshnnextcloud.go

```diff
@@ -41,3 +50,68 @@ func (c *concreteNextcloudProvider) ListVSHNnextcloud(ctx context.Context, names
 
 	return instances, nil
 }
+
+func (k *concreteNextcloudProvider) GetKubeConfig(ctx context.Context, instance vshnv1.VSHNNextcloud) ([]byte, error) {
```

Comment:
This method seems to be duplicated

Reply:
I've refactored this whole part and created a single method for all services